trends_presentation/src/commonMain/kotlin/net/thechance/mena/trends/presentation/screen/upload_reel/UploadReelViewModel.kt

```diff
@@ -196,12 +205,41 @@ internal class UploadReelViewModel(
         tryToExecute(
             block = { state.value.reelId?.let { reelsRepository.deleteReelById(id = it) } },
             onSuccess = { updateState { UploadReelScreenState() } },
-            onError = { errorState -> updateState { copy(errorState = errorState) } },
+            onError = { errorState -> onClickDeleteVideoError(errorState) },
```

Comment:
I think not need to this function only become : 
```suggestion
            onError = { errorState -> mapToUploadReelError(errorState) },
```

Reply:
```
  private fun mapToUploadReelError(errorState: ErrorState): UploadReelErrorState {
        return when (errorState) {
            is UploadReelErrorState -> errorState
            else -> UploadReelErrorState.RequestTimeout
        }
    }
```
    
    it is a mapper dosenot have a update state
    
```
       private fun onClickDeleteVideoError(errorState: ErrorState) {
        val uploadReelError = mapToUploadReelError(errorState)
        updateState { copy(errorState = uploadReelError) }
    }
```

Reply:
sorry but I don't understand you

Reply:
i its not okey to do your suggeston becase these function  dose not do the same thing 
one of theme have to updateState